Jest 20: mapCoverage

- Issue

With `mapCoverage` set to `true`, I seem to get this:

![code_coverage_report_for_all_files](https://cloud.githubusercontent.com/assets/188038/25792957/1a5ad5c2-340d-11e7-9bc8-cbead990a284.png)

in the reports with:

![code_coverage_report_for_src_base_serialization_file__users_joscha_dev_canva_web_src_base_serialization_deserialization_ts](https://cloud.githubusercontent.com/assets/188038/25792997/489fd252-340d-11e7-9ff0-22bfa50b4ffd.png)

when I drill into the file.

If I disable `mapCoverage` everything seems to work as expected regarding the coverage, but the mapping is shifted.

- Expected behavior

No `file://` links

not yet, it seems to have something to do with:
```
  "modulePaths": ["<rootDir>/src"],
```

Before Jest 20 I was able to just have:
```
  "roots": ["<rootDir>/src"],
```

but after upgrading everything that had a relative import `import x from 'bla/foo';` seemed to fail:
```
  ● Test suite failed to run

    Cannot find module 'base/serialization' from 'fixtures_proto.ts'

      at Resolver.resolveModule (node_modules/jest-resolve/build/index.js:172:17)
      at Object.<anonymous> (src/base/serialization/tests/fixtures/fixtures_proto.ts:5:25)
```
where it worked with Jest 19, so I changed `roots` to `modulePaths`, e.g. now I have:
```
  "modulePaths": ["<rootDir>/src"],
```

adding `roots` additionally doesn't seem to change anything.

--- a/README.md
+++ b/README.md
@@ -58,6 +58,8 @@ Prior to version `20.0.0`, coverage reports could be obtained using the inbuilt 
 
 To generate coverage results, set the `mapCoverage` property in the `jest` configuration section to `true`.
 
+> Please note that the `outDir` property in the `jest` configuration section is removed in coverage mode, due to [#201](https://github.com/kulshekhar/ts-jest/issues/201).
+
 ### React Native
 
 There is a few additional steps if you want to use it with React Native.

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "ts-jest",
-  "version": "20.0.0",
+  "version": "20.0.1",
   "main": "index.js",
   "types": "./dist/index.d.ts",
   "description": "A preprocessor with sourcemap support to help use Typescript with Jest",

--- a/src/utils.ts
+++ b/src/utils.ts
@@ -104,6 +104,12 @@ export function getTSConfig(globals, collectCoverage: boolean = false) {
 
     config.inlineSourceMap = true;
     config.inlineSources = true;
+
+    if (config.outDir) {
+      // the coverage report is broken if `.outDir` is set
+      // see https://github.com/kulshekhar/ts-jest/issues/201
+      delete config.outDir;
+    }
   }
 
   if (config.allowSyntheticDefaultImports) {

--- a/src/utils.ts
+++ b/src/utils.ts
@@ -98,18 +98,14 @@ export function getTSConfig(globals, collectCoverage: boolean = false) {
 
   //inline source with source map for remapping coverage
   if (collectCoverage) {
-    if (config.sourceMap) {
-      delete config.sourceMap;
-    }
+    delete config.sourceMap;
 
     config.inlineSourceMap = true;
     config.inlineSources = true;
 
-    if (config.outDir) {
-      // the coverage report is broken if `.outDir` is set
-      // see https://github.com/kulshekhar/ts-jest/issues/201
-      delete config.outDir;
-    }
+    // the coverage report is broken if `.outDir` is set
+    // see https://github.com/kulshekhar/ts-jest/issues/201
+    delete config.outDir;
   }
 
   if (config.allowSyntheticDefaultImports) {

